### PR TITLE
node:https: fix prototype chain of Agent

### PR DESCRIPTION
**Pull request by nektro**
fixes https://github.com/oven-sh/bun/issues/14184
fixed a few others while i was here

code

```js
import { request } from "node:https";

request("https://example.com/", { agent: false }).end();
```

before

```
❯ bun index.js 
ConnectionRefused: Unable to connect. Is the computer able to access the url?
 path: "https://example.com:80/"

Bun v1.1.33 (macOS arm64)
```

after

```
❯ ~/src/bun2/build/debug/bun-debug index.js 
```

### CODE REVIEW

**Comment by robobun** [comment]
@nektro, your commit bce5f5838b5058b5d3ab1cfcfda2aa31e089a508 is building: [`#6149`](https://buildkite.com/bun/bun/builds/6149)

File: test/js/node/http/node-http.test.ts [hunk 1]
```diff
@@ -2390,3 +2390,7 @@ it("must set headersSent to true after headers are sent when using chunk encoded
     server.close();
   }
 });
+
+it("should work when sending https.request with agent:false", () => {
+  https.request("https://example.com/", { agent: false }).end();
```
Comment (cirospaciari) [inline test/js/node/http/node-http.test.ts:2395]: this test dont fail in revision `1.1.34+5e5e7c60f`

**Review by cirospaciari — CHANGES_REQUESTED** [review]
Looks good but we should tweak the test to fail looks like is passing when I run it.

Screenshot 2024-11-14 at 18 12 41

File: test/js/node/http/node-http.test.ts [hunk 2]
```diff
@@ -2390,3 +2390,7 @@ it("must set headersSent to true after headers are sent when using chunk encoded
...
+});
```
Comment (cirospaciari) [inline test/js/node/http/node-http.test.ts]:
```suggestion
it("should work when sending https.request with agent:false", async () => {
  const { promise, resolve, reject } = Promise.withResolvers();
  const client = https.request("https://example.com/", { agent: false });
  client.on("error", reject);
  client.on("close", resolve);
  client.end();
  await promise;
});

```

**Review by cirospaciari — APPROVED** [review]
lets merge after CI

**Pull request merged**